### PR TITLE
Store payment info

### PR DESCRIPTION
**Pull request by andrei-21**

### CODE REVIEW

File: src/data_store.rs [hunk 1]
```diff
@@ -70,6 +128,39 @@ impl DataStore {
     }
 }
 
+// Store all provided exchange rates.
+// For every row it takes ~13 bytes (4 + 3 + 2 + 4), if we have 100 fiat currencies it adds 1300 bytes.
+// For 1000 payments it will add ~1 MB.
+fn insert_snapshot(
```
Comment (gcomte) [inline src/data_store.rs:134]: not very idiomatic function name. Isn't it a snapshot of exchange rates? How about `snapshot_exchange_rates`? Or a shorter version would be `snap_rates`. Or any combination of this.

File: src/lib.rs [hunk 2]
```diff
@@ -497,6 +527,24 @@ impl LightningNode {
...
+                        &invoice.payment_hash,
+                        user_preferences,
+                        exchange_rates,
+                        None,
+                    )
+                    .map_to_permanent_failure("Failed to persist payment info")
+            }
+            _ => Err(invalid_input("Invalid invoice")),
+        }?;
+
```
Comment (gcomte) [inline src/lib.rs:547]: `pay_invoice()` and `pay_open_invoice()` now duplicate a lot of code. We should probably create a private function that contains that code and is being called by `pay_invoice()` and `pay_open_invoice()`. Maybe in another PR.

File: src/data_store.rs (hunk 1 above)
Comment (danielgranhao) [reply src/data_store.rs]: Sure. Changed in [a761e80](https://github.com/getlipa/lipa-lightning-lib/pull/628/commits/a761e80dc06674ed86422f8537484ce3d8330912)

File: src/lib.rs (hunk 2 above)
Comment (danielgranhao) [reply src/lib.rs]: There's still some duplicated code, but it got improved in [c90d94e](https://github.com/getlipa/lipa-lightning-lib/pull/628/commits/c90d94ef4ad97535821a2a4f9fb5af7259137126). I would leave more improvements, if any, for later PRs.

**Review by gcomte — APPROVED** [review]

**Pull request merged**